Data/ByteString/Lazy/Internal.hs

```diff
@@ -158,15 +182,19 @@ unpackChars (Chunk c cs) = S.unpackAppendCharsLazy c (unpackChars cs)
 
 ------------------------------------------------------------------------
 
+-- We no longer use these invariant-checking functions internally,
+-- preferring an assertion on `Chunk` itself, controlled by the
+-- HS_BYTESTRING_ASSERTIONS preprocessor macro.
+
 -- | The data type invariant:
 -- Every ByteString is either 'Empty' or consists of non-null 'S.ByteString's.
--- All functions must preserve this, and the QC properties must check this.
+-- All functions must preserve this.
 --
 invariant :: ByteString -> Bool
```

Comment:
Let's mark it as deprecated then, so that we can remove it in the next major release.

Reply:
I'm not sure. It's very low-cost to keep these around. (Although perhaps also pretty low-value...)